### PR TITLE
Clean up unnecessary arguments in the codegen stage

### PR DESCRIPTION
**Pull request by EmilyBourne**
The codegen stage contains lots of old code which sometimes refers to parameters which no longer exist. This PR cleans up some of that code and removes commented dead code which refers to it. Specifically the functions `fcode`, `ccode` and `pycode` are removed as well as `Codegen.doprint` and `Codegen.set_printer`. An instance of `Codegen` must now be created with the language passed as argument.

### CODE REVIEW

**Review by yguclu — CHANGES_REQUESTED** [review]
Looks good to me!

I have just two minor comments.

File: pyccel/codegen/printing/ccode.py [hunk 1]
```diff
@@ -2515,27 +2515,3 @@ def indent_code(self, code):
             level += increase[n]
         return pretty
 
-def ccode(expr, filename, assign_to=None, **settings):
```
Comment (Farouk-Echaref) [inline pyccel/codegen/printing/ccode.py:2518]: what were the roles of these functions before? (ccode(), fcode(), pycode())

File: pyccel/codegen/codegen.py [hunk 2]
```diff
@@ -24,28 +23,30 @@
...
+    Class which handles the generation of code.
+
+    The class which handles the generation of code. This is done by creating an appropriate
+    class inheriting from `CodePrinter` and using it to create strings describing the code
+    that should be printed. This class then takes care of creating the necessary files.
+
+    Parameters
+    ----------
+    parser : SemanticParser
+        The Pyccel Semantic parser node.
```
[the remark this answers is not in the document]
Comment (EmilyBourne) [reply pyccel/codegen/codegen.py]: Fixed a2954f495

File: CHANGELOG.md [hunk 3]
```diff
@@ -28,6 +28,8 @@ All notable changes to this project will be documented in this file.
 -   \[INTERNALS\] Moved precision from `ast.basic.TypedAstNode` to an internal property of `ast.datatypes.FixedSizeNumericType` objects.
 -   \[INTERNALS\] Use cached `__add__` method to determine result type of arithmetic operations.
 -   \[INTERNALS\] Use cached `__and__` method to determine result type of bitwise comparison operations.
+-   \[INTERNALS\] Removed unused `Xcode` functions from printers.
```
[the remark this answers is not in the document]
Comment (EmilyBourne) [reply CHANGELOG.md]: Fixed 1e5f5507e

File: pyccel/codegen/printing/ccode.py (hunk 1 above)
Comment (Farouk-Echaref) [reply pyccel/codegen/printing/ccode.py:2518]: Clear enough, thanks for the clarification!

**Review by yguclu — APPROVED** [review]

**Pull request merged**